packages/botonic-plugin-contentful/src/contentful/contents/button.ts

```diff
@@ -32,7 +32,14 @@ export class ButtonDelivery {
   ): Promise<cms.Button> {
     // we could pass the entry to fromId to avoid fetching it again, but it makes
     // the code more complex when the reference is a button
-    return this.fromId(entry.sys.id, context)
+    try {
+      return await this.fromId(entry.sys.id, context)
```

Comment:
If the method is defined as `async`, is it necessary to `await` for the result? Is it double wrapping it into Promises?

Reply:
I only do the await when I use the result or then when want to treat the exceptions with a try/catch (through I'm not 100% about this one)

Reply:
@dpinol is right here. `return await` makes sense when you want to catch the rejection and return other stuff (and it only makes sense in that situation).
See https://jakearchibald.com/2017/await-vs-return-vs-return-await/